source/material_model/reactive_fluid_transport.cc

```diff
@@ -75,18 +165,44 @@ namespace aspect
                 // The fluid volume fraction in equilibrium with the solid
                 // at any point (stored in the melt_fractions vector) is
                 // equal to the sum of the bound fluid content and porosity.
-                const unsigned int bound_fluid_idx = this->introspection().compositional_index_for_name("bound_fluid");
                 melt_fractions[q] = in.composition[q][bound_fluid_idx] + in.composition[q][porosity_idx];
                 break;
               }
+              case tian_approximation:
+              {
+                // The bound fluid content is calculated using parametrized phase
+                // diagrams for four different rock types: sediment, MORB, gabbro, and
+                // peridotite.
+                const unsigned int sediment_idx = this->introspection().compositional_index_for_name("sediment");
+                const unsigned int MORB_idx = this->introspection().compositional_index_for_name("MORB");
+                const unsigned int gabbro_idx = this->introspection().compositional_index_for_name("gabbro");
+                const unsigned int peridotite_idx = this->introspection().compositional_index_for_name("peridotite");
+
+                // Initialize a vector that stores the compositions (mass fractions) for
+                // the four different rock compositions,
+                std::vector<double> tracked_rock_mass_fractions;
+                tracked_rock_mass_fractions.push_back(in.composition[q][peridotite_idx]);
+                tracked_rock_mass_fractions.push_back(in.composition[q][gabbro_idx]);
+                tracked_rock_mass_fractions.push_back(in.composition[q][MORB_idx]);
+                tracked_rock_mass_fractions.push_back(in.composition[q][sediment_idx]);
+
+                // The bound water content (water within the solid phase) for the four different rock types
+                std::vector<double> tian_eq_bound_water_content = tian_equilibrium_bound_water_content(in, q);
+
+                // average the water content between the four different rock types
+                double average_eq_bound_water_content = MaterialUtilities::average_value (tracked_rock_mass_fractions, tian_eq_bound_water_content, MaterialUtilities::arithmetic);
+
+                // The fluid volume fraction (stored in the melt_fractions vector) is equal to the porosity and the change
+                // in bound fluid content (current bound fluid - updated average bound fluid).
```

Comment:
```suggestion
                // The fluid volume fraction in equilibrium with the solid (stored in the melt_fractions vector) is equal to the sum of the porosity and the change
                // in bound fluid content (current bound fluid - updated average bound fluid).
```

Reply:
You can also describe this differently (which I would maybe find more intuitive):
The total water fraction is the sum of the current bound water fraction and the porosity. To get the free water fraction in equilibrium with the solid, we have to take this total water fraction and subtract the equilibrium bound water fraction from the Tian parametrization.